crates/holochain_sqlite/src/db.rs

```diff
@@ -270,6 +270,22 @@ impl<Kind: DbKindT> DbRead<Kind> {
             Err(e) => Err(DatabaseError::Timeout(e)),
         }
     }
+
+    #[cfg(any(test, feature = "test_utils"))]
+    pub fn test_read<R, F>(&self, f: F) -> R
```

Comment:
This gives us a sort-of-sync interface for calling the real database code. Provided for both read and write. You don't need to await and it tidies up the error handling a bit in tests but it has to be run on an existing Tokio runtime or you'll get an error.

Thoughts?

Reply:
I think that's what Michael was requesting earlier. `test_write` seems useful, and since `test_read` is the counterpart I believe it'll be equally useful.